carbon-design-system/carbon-addons-iot-react
# Empty state fix

Closes #2642 

**Summary**

Icons dynamically generated cannot contain ids or we get duplicated ids when icon is used more than once. Any of the empty state icons require ids in order to provide gradient for the fill. 

To fix the issue I have taken then and put the generated React Icon in a static folder inside of icon directory. 

**Change List (commits, features, bugs, etc)**

- Created a `useUniqueId` hook to provide a unique id for Icons
- moved all EmptyState React icon variations into a static directory,
- swapped all references to these components to new location
- added spy for hook so snapshot would not be different each time. 
- updated snapshots
- added a story to showcase same icon being used for multiple instances as seen in the issue

**Acceptance Test (how to verify the PR)**

- Make sure EmptyState stories all render 
- make sure any component that uses empty state (Table,List, etc) render correctly 





**Things to look for during review**
